health_monitor/src/health_monitor.cpp

```diff
@@ -139,8 +139,121 @@ namespace health_monitor
+            {
+             	nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_truck[6] = true;
+            } 
+            else if((crit_driver_status_truck=="s_0") && (is_published_truck[7] ==false)) //SSC Failure
+            {
+             	nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_truck[7] = true;
+            }       
+            
+        }//End truck
+        
+        else if (car_ == true)
+        {
+            if((crit_driver_status_car == "s_1_l_1_g_1") && (is_published_car[0] == false)) //"All essential drivers are ready"
+            {
+             	nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_car[0] = true;
+            } 
+            else if((start_time_flag_.isZero()==true) || (time_now - start_up_timestamp_ <= start_duration) && (is_published_car[1] == false)) //"System is starting up..."
+            {
+                nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_truck[1] = true;
+            }
+            else if((crit_driver_status_car == "s_1_l_1_g_0") && (is_published_car[2] == false)) //GPS Failure
+            {
+                nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+  		        is_published_car[2] = true;
+            }
+            else if((crit_driver_status_car == "s_1_l_0_g_1") && (is_published_car[3] == false)) //LIDAR Failure
+            {
+                nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_car[3] = true;
+            }
+            else if((crit_driver_status_car == "s_1_l_0_g_0") && (is_published_car[4] == false)) //LIDAR, GPS Failure
+            {
+                nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_car[4] = true;
+            }
+            else if((crit_driver_status_car == "s_0") && (is_published_car[5] == false)) //SSC Failure
+            {
+                nh_.publishSystemAlert(driver_manager_.handleSpin(truck_,car_,time_now,start_up_timestamp_,start_duration,start_time_flag_.isZero()));
+
+  		        is_published_car[5] = true;
+            }
+            
+
+        }//End car
+
+
         return true;
     }
 
+
+    void HealthMonitor::setisPublishedFalse()
+    {
+        for(int i=0; i<9; i++)
```

Comment:
use size_t for index types in for loops. Using signed types generally results in a warning. 
The size of these arrays should be constants to avoid typos. Or make the arrays vectors and use .size().

Reply:
If you use a std::vector you could also initialize to false with the constructor which would be a lot cleaner than these loops

std::vector my_vec(NumElements, InitialValue)